### DISCUSSION TITLE
How can I override Line Shader or Line Fill to create gradient lines?

### DISCUSSION THREAD

**Discussion by dmytroreutov** [patrykandpatrick/vico#908] [Questions]
### Question

Hello! I have requirement to draw line chart with folowing style:

<img width="705" alt="image" src="https://github.com/user-attachments/assets/1dc47935-ca8e-49b6-af42-cfeffa508b25">

Is it possible to override/customize `SingleLineFill` or `DynamicShader` for `LineCartesianLayer` to create something similar? Maybe I can rewrite default behaviour.

### Vico version(s)

2.0.0-alpha.28

### UI framework(s)

Views

**Comment by Gowsky** [comment]
Hello, @dmytroreutov. You can add this kind of glow (shadow) by using a custom implementation of `LineCartesianLayer.AreaFill`.

**Reply by dmytroreutov** [reply]
Hi! Thank you for your response.

And what about gradient for line fill. Is it actually possible to customise line with `DynamicShader` (or something) so I can set startColor/endColor for line `Paint`

**Reply by Gowsky** [reply]
I'm reposting my response because I misunderstood your question.

Yes, you can customize the line gradient using `LineFill` with `DynamicShader`. Please see [`DynamicShader.horizontalGradient`](https://patrykandpatrick.com/vico/api/2.0.0-beta.1/core/com.patrykandpatrick.vico.core.common.shader/-dynamic-shader/-companion/horizontal-gradient).
